packages/scandipwa/src/component/ProductCard/ProductCard.container.js

```diff
@@ -87,16 +93,29 @@ export class ProductCardContainer extends PureComponent {
     });
 
     _getLinkTo() {
-        const { product: { url }, product } = this.props;
+        const {
+            base_link_url,
+            product_use_categories,
+            product: { url, url_rewrites },
+            product
+        } = this.props;
+        const { pathname: storePrefix } = new URL(base_link_url || window.location.origin);
+        const { location: { pathname } } = history;
 
         if (!url) {
             return undefined;
         }
 
         const { parameters } = this._getConfigurableParameters();
 
+        const productUrl = `${pathname.replace(storePrefix, '')}/${url.replace(storePrefix, '')}`;
+
+        // eslint-disable-next-line fp/no-let
+        let rewriteUrl = url_rewrites.filter((e) => (e.url.includes(productUrl) ? e.url : null));
```

Comment:
No-let is a strict policy, please make this const. Also, please comment on this logic - it is not straightforward.